api/src/controllers/user.controller.js

```diff
@@ -1,51 +1,81 @@
-import jwt from 'jsonwebtoken';
 import bcrypt from 'bcryptjs';
 import UserService from '../services/user.service';
 
-const secret = process.env.SECRET || 'supersecret';
 
 const UserController = {
-  fetchAllUsers(req, res) {
-    const allUsers = UserService.fetchAllUsers();
-    // eslint-disable-next-line array-callback-return
-    allUsers.map((userObj) => {
-      Object.defineProperty(userObj, 'password', {
-        enumerable: false,
-        writable: true,
-      });
-    });
-    return res.json({
-      status: 200,
-      data: allUsers,
-    }).status(200);
-  },
+  // fetchAllUsers(req, res) {
+  //   const allUsers = UserService.fetchAllUsers();
+  //   // eslint-disable-next-line array-callback-return
+  //   allUsers.map((userObj) => {
+  //     Object.defineProperty(userObj, 'password', {
+  //       enumerable: false,
+  //       writable: true,
+  //     });
+  //   });
+  //   return res.json({
+  //     status: 200,
+  //     data: allUsers,
+  //   }).status(200);
+  // },
 
-  addAUser(req, res) {
+  async addAUser(req, res) {
     const newUser = req.body;
     if (!newUser.type && !newUser.isAdmin) {
       newUser.type = 'client';
       newUser.isAdmin = false;
     }
+    if (!newUser.firstName) {
+      return res.json({
+        status: 400,
+        error: 'First Name field is required',
+      });
+    }
+    if (!newUser.lastName) {
+      return res.json({
+        status: 400,
+        error: 'Last Name field is required',
+      });
+    }
+    if (!newUser.email) {
+      return res.json({
+        status: 400,
+        error: 'Email field is required',
+      });
+    }
+    if (!newUser.password) {
+      return res.json({
+        status: 400,
+        error: 'Password field is required',
+      });
+    }
     const hashedPassword = bcrypt.hashSync(newUser.password, 8);
     newUser.password = hashedPassword;
-    const createdUser = UserService.addUser(newUser);
-    const token = jwt.sign({ id: createdUser.id }, secret, {
-      expiresIn: 86400,
-    });
-    createdUser.token = token;
+    const createdUser = await UserService.addUser(newUser);
     Object.defineProperty(createdUser, 'password', {
       enumerable: false,
       writable: true,
     });
     return res.json({
       status: 201,
       data: createdUser,
-    }).status(201);
+    });
   },
 
-  signIn(req, res) {
+  async signIn(req, res) {
     const oldUser = req.body;
-    const foundUser = UserService.signIn(oldUser);
+    if (!oldUser.email) {
+      return res.json({
+        status: 400,
+        error: 'Email field is required',
```

Comment:
same here. write a validator middleware and call it on the route

Reply:
Done. I used joi